pkg/signal/signal.go

```diff
@@ -4,95 +4,77 @@
 // Package signal provides types for working with feedback signal.
 package signal
 
-type (
-	elemType uint64
-	prioType int8
+import (
+	"maps"
 )
 
-type Signal map[elemType]prioType
+type Signal struct {
+	m map[uint64]int8
+}
+
+var Nil = Signal{}
+
+func makeSignal(n ...int) Signal {
+	if len(n) == 0 {
+		return Signal{
+			m: make(map[uint64]int8),
+		}
+	}
+	return Signal{
+		m: make(map[uint64]int8, n[0]),
+	}
+}
 
 func (s Signal) Len() int {
-	return len(s)
+	return len(s.m)
 }
 
-func (s Signal) Empty() bool {
-	return len(s) == 0
+func (s Signal) Nil() bool {
```

Comment:
There is one caller for this. Do we really need to distinguish nil vs empty? That's usually error-prone. If not, let's keep only empty.

Reply:
```golang
	if !r.SignalFilter.Nil() && !collectSignal {
		return fmt.Errorf("SignalFilter must be used with FlagCollectSignal")
	}
```
is the only SignalFilter reader I see. There are no writers.
@a-nogikh it looks safe to remove this field declaration from `fuzzer.queue.Request`.